clar2wasm/src/words/sequences.rs

```diff
@@ -1354,4 +1567,153 @@ mod tests {
+                    .unwrap()
+            )
+        );
+    }
+
+    #[test]
+    fn test_map_simple_buff() {
+        assert_eq!(
+            eval(
+                r#"
+(define-private (zero-or-one (char (buff 1))) (if (is-eq char 0x00) 0x00 0x01))
+(map zero-or-one 0x000102) 
+        "#
+            ),
+            Some(
+                Value::cons_list_unsanitized(vec![
+                    Value::buff_from_byte(0),
+                    Value::buff_from_byte(1),
+                    Value::buff_from_byte(1)
+                ])
+                .unwrap()
+            )
+        );
+    }
+
+    #[test]
+    fn test_map_simple_string_ascii() {
+        assert_eq!(
+            eval(
+                r#"
+(define-private (a-or-b (char (string-ascii 1))) (if (is-eq char "a") "a" "b"))
+(map a-or-b "aca")
+        "#
+            ),
+            Some(
+                Value::cons_list_unsanitized(vec![
+                    Value::string_ascii_from_bytes(vec![0x61]).unwrap(),
+                    Value::string_ascii_from_bytes(vec![0x62]).unwrap(),
+                    Value::string_ascii_from_bytes(vec![0x61]).unwrap(),
+                ])
+                .unwrap()
+            )
+        );
+    }
+
+    #[test]
+    fn test_map_simple_string_utf8() {
+        assert_eq!(
+            eval(
+                r#"
+(define-private (a-or-b (char (string-utf8 1))) (if (is-eq char u"a") u"a" u"b"))
+(map a-or-b u"aca")
+        "#
+            ),
+            Some(
+                Value::cons_list_unsanitized(vec![
+                    Value::string_utf8_from_bytes(vec![0x61]).unwrap(),
+                    Value::string_utf8_from_bytes(vec![0x62]).unwrap(),
+                    Value::string_utf8_from_bytes(vec![0x61]).unwrap(),
+                ])
+                .unwrap()
+            )
+        );
+    }
+
+    // TODO: The string-utf8 can be uncommented when #216 is merged.
+    #[test]
+    fn test_map_mixed() {
+        assert_eq!(
+            eval(
+                r#"
+(define-private (add-everything
+    (a int)
+    (b uint)
+    (c (string-ascii 1))
+    ;;(d (string-utf8 1))
+    (e (buff 1))
+    )
+    (+
+        a
+        (to-int b)
+        (unwrap-panic (string-to-int? c))
+        ;;(unwrap-panic (string-to-int? d))
+        (buff-to-int-be e)
+    )
+)
+(map add-everything
+    (list 1 2 3)
+    (list u1 u2 u3)
+    "123"
+    ;;u"123"
+    0x010203
+)
+        "#
+            ),
+            Some(
+                Value::cons_list_unsanitized(vec![Value::Int(4), Value::Int(8), Value::Int(12),])
+                    // Value::cons_list_unsanitized(vec![Value::Int(5), Value::Int(10), Value::Int(15),])
+                    .unwrap()
+            )
+        );
+    }
+
+    #[test]
+    fn test_map() {
+        const MAP_FNS: &str = "
+(define-private (addify-1 (a int))
+  (+ a 1))
+
+(define-private (addify-2 (a int) (b int))
+  (+ a b 1))
+";
+
+        // let a = &format!("{MAP_FNS} (map addify-1 (list 1 2 3))");
```

Comment:
commented out test should be uncommented or removed

Reply:
Fixed in fcb2c79.